### PR TITLE
Pluribus Networks pn access list ip module

### PR DESCRIPTION
**Pull request by rajaspachipulusu17**
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
pn_access_list_ip module  which would add associate ip address to access list.

##### ADDITIONAL INFORMATION
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

### CODE REVIEW

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+
+DOCUMENTATION = """
+---
+module: pn_access_list_ip
+author: "Pluribus Networks (rpachi@pluribusnetworks.com)"
+version_added: "2.8"
+short_description: CLI command to add/remove access-list-ip
+description:
+  - This modules is used to dd IPs associated with access list
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list_ip.py]:
```suggestion
  - This modules is used to add IPs associated with access list
```

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    type: str
+  state:
+    description:
+      - State the action to perform. Use 'present' to add access-list-ip and
+      'absent' to remove access-list-ip.
+    required: True
+    choices: [ "present", "absent"]
+  pn_ip:
+    description:
+      - IP associated with the access list
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list_ip.py]:
```suggestion
      - IP associated with the access list.
```

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    required: false
+    type: str
+    default: '::'
+  pn_name:
+    description:
+      - Access List Name
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list_ip.py]:
```suggestion
      - Access List Name.
```

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    This method checks for idempotency using the access-list-ip-show command.
+    If ip  exists, return IP_EXISTS as True else False.
+    :param module: The Ansible module to fetch input parameters
+    :param cli: The CLI string
+    :return Global Booleans: IP_EXISTS
+    """
+    name = module.params['pn_name']
+    ip = module.params['pn_ip']
+
+    show = cli + \
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list_ip.py]:
```python
show = [cli, 'access-list-ip-show', 'name', name, 'format', 'ip', 'no-show-headers'] 
out = module.run_command(show, use_unsafe_shell=True)[1]
```

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  description: set of error responses from the access-list-ip command.
+  returned: on error
+  type: list
+changed:
+  description: indicates whether the CLI caused changes on the target.
+  returned: always
+  type: bool
+"""
+
+import shlex
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_access_list_ip.py]: `run_command` internally uses shlex. You need to pass command as list and use `use_unsafe_shell=True`. You can find the documentation here - https://github.com/ansible/ansible/blob/68cafbda2baa071dce2efcd298463971d92a32e3/lib/ansible/module_utils/basic.py#L2702

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@rajaspachipulusu17 Could you please add test case for this change ? Thanks.

File: lib/ansible/modules/network/netvisor/pn_access_list_ip.py (hunk 4 above)
Comment (rajaspachipulusu17) [reply lib/ansible/modules/network/netvisor/pn_access_list_ip.py]: ok let change and test and send it for review.

Comment (rajaspachipulusu17) [reply lib/ansible/modules/network/netvisor/pn_access_list_ip.py]: Above show which you have suggested wont split my cli parameter.

Hence i have concatenated the string before using it in run_command.

  out = module.run_command(cli.split(), use_unsafe_shell=True)[1]

**Comment by rajaspachipulusu17** [comment]
@Akasurde 

Unit test cases are ready. Should i create seperate PR or can i send it in same PR?

Thanks,
Rajas

**Comment by Akasurde** [comment]
@rajaspachipulusu17 if unit tests are related `pn_access_list_ip.py` then in this same PR. Or else in a separate PR. Thanks.

**Comment by rajaspachipulusu17** [comment]
closing this PR, as i deleted branch accidentally. new PR is  49604

**Comment by abadger** [comment]
Closing in favour of #49604

**Pull request closed without merging**